Unable to install spectral-language-server

### I've made an initial investigation into the error myself

- [ ] Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

Due to changes in upstream package, it's fails to install.


### Neovim version (>= 0.7)

NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3


### Operating system/version

Linux laptop 5.19.10-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Tue, 20 Sep 2022 15:18:01 +0000 x86_64 GNU/Linux

### Affected packages

spectral-language-server

### Actual behavior

npm fails to install language server
It also fails when install it via npm directly to the  system as author made a new version of lsp. 

### Expected behavior

Installed package

### Mason output

```Text
◍ spectral-language-server
      Wrote to /home/alex/.local/share/nvim/mason/.packages/spectral-language-server/package.json:
      
      {
        "name": "@mason/spectral-language-server",
        "version": "1.0.0",
        "description": "",
        "main": "index.js",
        "scripts": {
          "test": "echo \"Error: no test specified\" && exit 1"
        },
        "keywords": [],
        "author": "",
        "license": "ISC"
      }
      
      
      npm WARN deprecated @stoplight/spectral@5.9.2: Spectral's latest version is now available as @stoplight/spectral-cli on npm
      npm ERR! code 127
      npm ERR! path /home/alex/.local/share/nvim/mason/.packages/spectral-language-server/node_modules/spectral-language-server
      npm ERR! command failed
      npm ERR! command sh -c -- ./script/vscode-spectral-build.sh
      npm ERR! sh: line 1: ./script/vscode-spectral-build.sh: No such file or directory
      
      npm ERR! A complete log of this run can be found in:
      npm ERR!     /home/alex/.npm/_logs/2022-09-25T11_09_09_405Z-debug-0.log
      spawn: npm failed with exit code 127 and signal 0.
```


### Installation log

```Text
[INFO  Sun 25 Sep 2022 02:09:08 PM MSK] ...acker/start/mason.nvim/lua/mason-core/installer/init.lua:110: Executing installer for Package(name=spectral-language-server)
[ERROR Sun 25 Sep 2022 02:09:25 PM MSK] ...acker/start/mason.nvim/lua/mason-core/installer/init.lua:146: Installation failed for Package(name=spectral-language-server) error=spawn: npm failed with exit code 127 and signal 0.
```


### Healthcheck

```Text
mason: require("mason.health").check()
========================================================================
## mason.nvim report
  - OK: neovim version >= 0.7.0
  - OK: **Go**: `go version go1.19.1 linux/amd64`
  - WARNING: **cargo**: not available
  - WARNING: **luarocks**: not available
  - OK: **Ruby**: `ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-linux]`
  - OK: **RubyGem**: `3.1.6`
  - WARNING: **Composer**: not available
  - OK: **PHP**: `PHP 8.1.10 (cli) (built: Sep  3 2022 04:38:10) (NTS)`
  - OK: **npm**: `8.19.2`
  - OK: **node**: `v18.9.0`
  - OK: **python3**: `Python 3.10.7`
  - OK: **pip3**: `pip 22.2.2 from /usr/lib/python3.10/site-packages/pip (python 3.10)`
  - OK: **javac**: `javac 11.0.16.1`
  - OK: **java**: `openjdk version "11.0.16.1" 2022-08-12`
  - WARNING: **julia**: not available
  - OK: **wget**: `GNU Wget 1.21.3 built on linux-gnu.`
  - OK: **curl**: `curl 7.85.0 (x86_64-pc-linux-gnu) libcurl/7.85.0 OpenSSL/1.1.1q zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.3 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.49.0`
  - OK: **gzip**: `gzip 1.12`
  - OK: **tar**: `tar (GNU tar) 1.34`
  - OK: **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
  - OK: **sh**: `Ok`
  - OK: GitHub API rate limit. Used: 12. Remaining: 48. Limit: 60. Reset: Sun 25 Sep 2022 02:34:21 PM MSK.
```


### Screenshots

_No response_

--- a/lua/mason-core/managers/github/init.lua
+++ b/lua/mason-core/managers/github/init.lua
@@ -18,7 +18,6 @@ local M = {}
 ---@param asset_file string
 ---@param release string
 local function with_release_file_receipt(repo, asset_file, release)
-    ---@return InstallReceiptGitHubReleaseFileSource
     return function()
         local ctx = installer.context()
         ctx.receipt:with_primary_source {

--- a/lua/mason-registry/spectral-language-server/init.lua
+++ b/lua/mason-registry/spectral-language-server/init.lua
@@ -1,6 +1,9 @@
 local Pkg = require "mason-core.package"
-local npm = require "mason-core.managers.npm"
+local git = require "mason-core.managers.git"
+local github = require "mason-core.managers.github"
 local _ = require "mason-core.functional"
+local Optional = require "mason-core.optional"
+local path = require "mason-core.path"
 
 return Pkg.new {
     name = "spectral-language-server",
@@ -11,5 +14,19 @@ return Pkg.new {
     homepage = "https://github.com/luizcorreia/spectral-language-server",
     languages = { Pkg.Lang.JSON, Pkg.Lang.YAML },
     categories = { Pkg.Cat.LSP },
-    install = npm.packages { "spectral-language-server", bin = { "spectral-language-server" } },
+    ---@async
+    ---@param ctx InstallContext
+    install = function(ctx)
+        local source = github.tag { repo = "stoplightio/vscode-spectral" }
+        source.with_receipt()
+        ctx.fs:mkdir "build"
+        ctx:chdir("build", function()
+            git.clone { "https://github.com/stoplightio/vscode-spectral", version = Optional.of(source.tag) }
+            ctx.spawn.npm { "install" }
+            ctx.spawn.node { "make", "package" }
+        end)
+        ctx.fs:rename(path.concat { "build", "dist", "server", "index.js" }, "spectral-language-server.js")
+        ctx.fs:rmrf "build"
+        ctx:write_node_exec_wrapper("spectral-language-server", "spectral-language-server.js")
+    end,
 }

